test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/Hubs.cs

```diff
@@ -37,7 +37,20 @@ public class TestHub : Hub
 
         public IEnumerable<string> GetHeaderValues(string[] headerNames)
         {
-            var headers = Context.Connection.GetHttpContext().Request.Headers;
+            var context = Context.Connection.GetHttpContext();
+
+            if (context == null)
+            {
+                throw new InvalidOperationException("Unable to get HttpContext from request.");
+            }
+
+            var headers = context.Request.Headers;
+
+            if (headers == null)
+            {
+                throw new InvalidOperationException("Unable to get headers from context.");
```

Comment:
When would this ever happen? Did you make a mock?

Reply:
Because I can't figure out why this test fails. Only long polling, only on OSX https://travis-ci.org/aspnet/SignalR/jobs/355210173#L1309 😢